### ISSUE TITLE
navigationOptions changes do not hot reload

### ISSUE THREAD

**Issue by BertCatsburg** [react-navigation/react-navigation#8725]
Exactly the same issue as #2623, but that one is closed almost 3 years ago, so I decided to create a new issue.

**Current Behavior**

- What code are you running and what is happening?

Standard expo init project, with minimal Navigator and HelloWorld-screen. 
See code below.

### App.js: 

```
import React, {useState} from 'react';
import {AppLoading} from 'expo';
import * as Font from 'expo-font'; // Font utilities
import {enableScreens} from 'react-native-screens';

import MainNavigator from './navigation/MainNavigator';

enableScreens();

const fetchFonts = () => {
    return Font.loadAsync({
        'regular': require('./assets/fonts/Roboto-Regular.ttf'),
        'bold': require('./assets/fonts/Roboto-Bold.ttf'),
        'medium': require('./assets/fonts/Roboto-Medium.ttf'),
    });
}

export default function App() {

    const [dataLoaded, setDataLoaded] = useState(false);

    if (!dataLoaded) {
        return <AppLoading
            startAsync={fetchFonts}
            onFinish={() => setDataLoaded(true)}
            onError={(err) => console.log(err)}
        />
    }

    return (
        <MainNavigator/>
    );
}

```

### MainNavigator.js
```
import React from 'react';
import {createStackNavigator} from 'react-navigation-stack';
import {createAppContainer} from 'react-navigation';


// Screens
import HelloWorldScreen from '../screens/HelloWorldScreen';


// *** Stack Navigator
const MainNavigator = createStackNavigator({
    HelloWorld: HelloWorldScreen
})

export default createAppContainer(MainNavigator);
```

### HelloWorldScreen.js

```
import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const HelloWorldScreen = props => {
    return (
        <View style={styles.screen}>
            <Text>Hello World</Text>
        </View>
    )
}

HelloWorldScreen.navigationOptions = {
    headerTitle: 'Hi Mom'
}

const styles = StyleSheet.create({
    screen: {
        flex: 1,
        width: '100%',
        justifyContent: 'flex-start',
        alignItems: 'center',
    }
})

export default HelloWorldScreen

```

### package.json
```
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@react-native-community/masked-view": "0.1.10",
    "expo": "~38.0.8",
    "expo-font": "~8.2.1",
    "expo-status-bar": "^1.0.2",
    "react": "~16.11.0",
    "react-dom": "~16.11.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz",
    "react-native-gesture-handler": "~1.6.0",
    "react-native-material-ui": "^1.30.1",
    "react-native-paper": "^4.0.1",
    "react-native-reanimated": "~1.9.0",
    "react-native-safe-area-context": "~3.0.7",
    "react-native-screens": "~2.9.0",
    "react-native-web": "~0.11.7",
    "react-navigation": "^4.4.0",
    "react-navigation-drawer": "^2.5.0",
    "react-navigation-material-bottom-tabs": "^2.3.0",
    "react-navigation-stack": "^2.8.2",
    "react-navigation-tabs": "^2.9.0"
  },
  "devDependencies": {
    "@babel/core": "^7.8.6",
    "babel-preset-expo": "~8.1.0"
  },
  "private": true
}

```

**Expected Behavior**

- What do you expect should be happening?

When changing the headerTitle in HelloWorldScreen.js I would expect that my connected phone shows the new title after a hot reload. It does the Refresh (hot-reload?) but does not reflect the change. I need to shake-reload to see the change.

- Include a screenshot or video if it makes sense.

**How to reproduce**

- Create standard expo reactnative project
- Use package.json and run: npm install
- Put the 3 files (App.js, MainNavigator.js and HelloWorldScreen.js) in the right places.
- Do an npm start
- Hold the iPhone-camera against the QR-code
- Make a change in the HelloWorldScreen.navigationOptions

You would expect the change to appear. But what happens is that the iPhone does refresh, but the change does not appear. 
Only after a shake-reload the change is shown.

**Your Environment**
- WebStorm IDE on osx Catalina
- iPhone

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | latest version
| node                           | v14.5.0 (on nvm)
| npm or yarn                    | 6.14.5
| all packages | See package.json

**Comment by satya164** [comment]
It's a limitation with how React Hot Loader and static properties work. It can't be fixed.
